pkg/controller/qdrouterd/qdrouterd_controller.go

```diff
@@ -231,6 +251,97 @@ func (r *ReconcileQdrouterd) Reconcile(request reconcile.Request) (reconcile.Res
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: instance.Name + "-selfsigned", Namespace: instance.Namespace}, selfSignedCertFound)
+		if err != nil && errors.IsNotFound(err) {
+			// Create a new self signed certificate
+			cert := certificates.NewSelfSignedCertificateForCR(instance)
+			controllerutil.SetControllerReference(instance, cert, r.scheme)
+			reqLogger.Info("Creating a new self signed cert %s%s\n", cert.Namespace, cert.Name)
+			err = r.client.Create(context.TODO(), cert)
+			if err != nil {
+				reqLogger.Info("Failed to create new self signed cert", "error", err)
+				return reconcile.Result{}, err
+			}
+			// Cert created successfully - return and requeue
+			return reconcile.Result{Requeue: true}, nil
+		} else if err != nil {
+			reqLogger.Info("Failed to create self signed cert", "error", err)
+			return reconcile.Result{}, err
+		}
+		caSecretName = selfSignedCertFound.Name
+	}
+
+    // If needed, check if CA issuer exists and if not create a new one
+	if requestCert {
+		caIssuerFound := &cmv1alpha1.Issuer{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: instance.Name + "-ca", Namespace: instance.Namespace}, caIssuerFound)
+		if err != nil && errors.IsNotFound(err) {
+			// Define a new ca issuer
+			newIssuer := certificates.NewCAIssuerForCR(instance, caSecretName)
+			controllerutil.SetControllerReference(instance, newIssuer, r.scheme)
+			reqLogger.Info("Creating a new ca issuer %s%s\n", newIssuer.Namespace, newIssuer.Name)
+			err = r.client.Create(context.TODO(), newIssuer)
+			if err != nil {
+				reqLogger.Info("Failed to create new ca issuer", "error", err)
+				return reconcile.Result{}, err
+			}
+			// Issuer created successfully - return and requeue
+			return reconcile.Result{Requeue: true}, nil
+		} else if err != nil {
+			reqLogger.Info("Failed to get ca issuer", "error", err)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// If needed, create certs for SslProfiles
+	if requestCert {
+		for _, profile := range instance.Spec.SslProfiles {
+			caCertFound := &cmv1alpha1.Certificate{}
+			err = r.client.Get(context.TODO(), types.NamespacedName{Name: instance.Name + "-" + profile.Name, Namespace: instance.Namespace}, caCertFound)
+			if err != nil && errors.IsNotFound(err) {
+				// Create a new certificate
+				cert := certificates.NewCACertificateForCR(instance, profile.Name)
+				controllerutil.SetControllerReference(instance, cert, r.scheme)
+				reqLogger.Info("Creating a new ca cert %s%s\n", cert.Namespace, cert.Name)
+				err = r.client.Create(context.TODO(), cert)
+				if err != nil {
+					reqLogger.Info("Failed to create new ca cert", "error", err)
+					return reconcile.Result{}, err
+				}
+				// Cert created successfully - return and requeue
+				return reconcile.Result{Requeue: true}, nil
+			} else if err != nil {
+				reqLogger.Info("Failed to create ca cert", "error", err)
+				return reconcile.Result{}, err
+			}
```

Comment:
Is there a reason to return after every creation attempt, rathe rthan continuing with the next thing to create immediately? (Possibly releated, is there a reason to repeat the if clause with the identical condition though the condition won't be changed by the first conditional block?)

I may be missing something but on first reading it seems that the following flow would be easier to read?
if requestCert {
   if instance.spec.Issuer == "" {
       //create issuer if needed
       //if failed return with error
       //create self signed ca if needed
       //if failed return with error
   }
   //create ca issuer if needed using our self signed cert 
   //if failed return with error
   //create certificates as needed for ssl profiles that do not specify the necessary secrets
}